x-pack/plugins/aiops/public/components/log_categorization/sampling_menu/sampling_menu.tsx

```diff
@@ -0,0 +1,185 @@
+
+import { RandomSamplerRangeSlider } from './random_sampler_range_slider';
+import {
+  RandomSampler,
+  RandomSamplerOption,
+  RANDOM_SAMPLER_OPTION,
+  RANDOM_SAMPLER_SELECT_OPTIONS,
+} from './random_sampler';
+
+interface Props {
+  randomSampler: RandomSampler;
+  reload: () => void;
+}
+
+export const SamplingMenu: FC<Props> = ({ randomSampler, reload }) => {
+  const [showSamplingOptionsPopover, setShowSamplingOptionsPopover] = useState(false);
+
+  const samplingProbability = useObservable(
+    randomSampler.getProbability$(),
+    randomSampler.getProbability()
+  );
+  const setSamplingProbability = useCallback(
+    (probability: number | null) => {
+      randomSampler.setProbability(probability);
+      reload();
+    },
+    [reload, randomSampler]
+  );
+
+  const randomSamplerPreference = useObservable(randomSampler.getMode$(), randomSampler.getMode());
+  const setRandomSamplerPreference = useCallback(
+    (mode: RandomSamplerOption) => {
+      randomSampler.setMode(mode);
+      reload();
+    },
+    [randomSampler, reload]
+  );
+
+  const { calloutInfoMessage, buttonText } = useMemo(() => {
+    switch (randomSamplerPreference) {
+      case RANDOM_SAMPLER_OPTION.OFF:
+        return {
+          calloutInfoMessage: i18n.translate(
+            'xpack.aiops.logCategorization.randomSamplerSettingsPopUp.offCallout.message',
+            {
+              defaultMessage:
+                'Random sampling can be turned on for the total document count and chart to increase speed although some accuracy will be lost.',
+            }
+          ),
+          buttonText: i18n.translate(
+            'xpack.aiops.logCategorization.randomSamplerSettingsPopUp.offCallout.button',
+            {
+              defaultMessage: 'No sampling',
+            }
+          ),
+        };
+      case RANDOM_SAMPLER_OPTION.ON_AUTOMATIC:
+        return {
+          calloutInfoMessage: i18n.translate(
+            'xpack.aiops.logCategorization.randomSamplerSettingsPopUp.onAutomaticCallout.message',
+            {
+              defaultMessage:
+                'The total document count and chart use random sampler aggregations. The probability is automatically set to balance accuracy and speed.',
+            }
+          ),
+          buttonText: i18n.translate(
+            'xpack.aiops.logCategorization.randomSamplerSettingsPopUp.onAutomaticCallout.button',
+            {
+              defaultMessage: 'Auto sampling',
+            }
+          ),
+        };
+
+      case RANDOM_SAMPLER_OPTION.ON_MANUAL:
+      default:
+        return {
+          calloutInfoMessage: i18n.translate(
+            'xpack.aiops.logCategorization.randomSamplerSettingsPopUp.onManualCallout.message',
+            {
+              defaultMessage:
+                'The total document count and chart use random sampler aggregations. A lower percentage probability increases performance, but some accuracy is lost.',
```

Comment:
As above, the first part of this message should be edited so that it only refers to the components in the flyout that are using random sampling.

Reply:
Updated in https://github.com/elastic/kibana/pull/153449/commits/8b3a2813af4b31d7567c263c80e723758ea13c3d